scripts/train/train.py

```diff
@@ -32,95 +32,131 @@
-            for loader in eval_loader:
-                if loader.label is None:
-                    raise ValueError(
-                        'When specifying multiple evaluation datasets, each one must include the \
+    # Check for missing mandatory fields
+    for field in TRAIN_CONFIG_KEYS:
+        _ = getattr(train_config, field)
+
+    # Validate the rest of the config
+    loaders = [train_config.train_loader]
+    if train_config.eval_loaders is not None:
+        for loader in (train_config.eval_loaders or []):  # pyright
+            if 'label' not in loader or loader['label'] is None:
+                raise ValueError(
+                    'When specifying multiple evaluation datasets, each one must include the \
                             `label` attribute.')
-                loaders.append(loader)
-        else:
-            loaders.append(eval_loader)
+            loaders.append(loader)
+    if train_config.eval_loader is not None:
+        loaders.append(train_config.eval_loader)
     for loader in loaders:
-        if loader.name == 'text':
-            if cfg.model.name == 'hf_t5':
+        if loader['name'] == 'text':
+            if train_config.model['name'] == 'hf_t5':
                 raise ValueError(
-                    f'Model type "{cfg.model.name}" is not supported when using the "text " ' +\
+                    f'Model type "{train_config.model["name"]}" is not supported when using the "text " ' +\
                     f'dataloader. Only finetuning is supported.')
 
-    if 'icl_tasks' in cfg:
-        if cfg.model.name == 'hf_t5':
+    if train_config.icl_tasks is not None or train_config.icl_tasks_str is not None:
+        if train_config.model['name'] == 'hf_t5':
             raise ValueError(
                 'ICL evaluation does not currently support Encoder-Decoder models, such as "hf_t5".'
             )
 
-    if (cfg.model.get('fc_type', 'torch') != 'te' and 'te' not in cfg.model.get(
-            'ffn_config', {}).get('ffn_type', 'mptmlp') and
-            'fp8' in cfg.precision):
+    if (train_config.model.get('fc_type', 'torch') != 'te' and
+            'te' not in train_config.model.get('ffn_config', {}).get(
+                'ffn_type', 'mptmlp') and 'fp8' in train_config.precision):
         warnings.warn(
             "fp8 only supported for te.Linear layers. Either set `cfg.model.fc_typ='te'` or "
             +
             "`cfg.model.ffn_config.ffn_type='te_ln_mlp'` to enable layers using fp8 precision."
         )
 
-    if (cfg.model.get('fc_type', 'torch') == 'te' or
-            'te' in cfg.model.get('ffn_config', {}).get('ffn_type', 'mptmlp')):
-        fsdp_config = cfg.get('fsdp_config', None)
-        act_ckpt = fsdp_config.get('activation_checkpointing', False)
+    if (train_config.model.get('fc_type', 'torch') == 'te' or
+            'te' in train_config.model.get('ffn_config', {}).get(
+                'ffn_type', 'mptmlp')):
+        if train_config.fsdp_config is None:
```

Comment:
I don't think this is the right behavior. FSDP config being None means that we don't want to use FSDP. So we don't want to force turn it on here